cmd/syncthing/monitor.go

```diff
@@ -212,19 +208,27 @@ func monitorMain(options serveOptions) {
 	}
 }
 
-func expandExecutableInCurrentDirectory(args0 string) (string, error) {
+func getBinary(args0 string) (string, error) {
+	e, err := os.Executable()
+	if err == nil {
+		return e, nil
+	}
+	// Check if args0 cuts it
+	_, err = exec.LookPath(args0)
+	if err == nil {
+		return args0, nil
+	}
 	// Works around a restriction added in go1.19 that executables in the
 	// current directory are not resolved when specifying just an executable
 	// name (like e.g. "syncthing")
 	if !strings.ContainsRune(args0, os.PathSeparator) {
-		// Check if it's in PATH
+		e = "." + string(os.PathSeparator) + args0
```

Comment:
I still think we should skip this entire thing, os.Executable() + exec.LookPath() should cover the reasonable scenarios imho. At least I don't see the scenario where it would be required or useful to also look separately in pwd, so despite not seeing how this would cause a problem by itself I don't think we should do it...

Reply:
Ok, done. My thinking was that `os.Executable` should anyway work all the time, but we want to try everything we can to get an executable. I guess we could even get rid of the `LookupPath` and just do `os.Executable` and fail if it returns an error?